src/transaction/SimulationResults.tsx

```diff
@@ -261,16 +264,70 @@ function createSimulationElements(simulationChecks:Map<string, Boolean>) {
 
 const SimulationResults = function SimulationResults() {
   const navigate: NavigateFunction = useNavigate();
+
   const onSendTransaction = async (txReq: TransactionRequest) => {
     const pendingTxStore = await UserState.getPendingTxStore();
-    const wallet = await grabWallet();
-    if (wallet === null) {
-      return;
-    }
+    try {
+      const wallet = await grabWallet();
+      if (wallet === null) {
+        const toastMsg = document.getElementById('toast-message');
+        toastMsg!.innerHTML = "Couldn't connect to wallet. Try resending the transaction.";
+
+        const toast = document.getElementById('errToast');
+        toast!.className = 'toast show';
+        setTimeout(() => {
+          toast!.className = 'toast hide';
+        }, 3000);
+      } else {
+        try {
+          const tResp: TransactionResponse = await wallet.sendTransaction(txReq);
+          await pendingTxStore.addPendingTransaction(tResp, true);
+          navigate('/Home');
+        } catch (err:any) {
+          const toastMsg = document.getElementById('toast-message');
+          const i = String(err).indexOf('(');
+          if (String(err).indexOf('bad response') !== -1) {
+            toastMsg!.innerHTML = 'The simulation ran into a network error. Please try testing the transaction again.';
+          } else if (String(err).indexOf(ERROR) !== -1 && i !== -1) {
+            let errMsg = String(err).substring(ERROR.length, i - 1);
+            errMsg = errMsg[0].toUpperCase().concat(errMsg.substring(1));
+            toastMsg!.innerHTML = errMsg;
+          } else if (String(err).indexOf(ERROR) !== -1 && i === -1) {
+            const errMsg = String(err).substring(ERROR.length);
+            toastMsg!.innerHTML = errMsg;
+          } else {
+            toastMsg!.innerHTML = err;
+          }
+
+          const toast = document.getElementById('errToast');
+          toast!.className = 'toast show';
+          setTimeout(() => {
+            toast!.className = 'toast hide';
+          }, 3000);
+        }
+      }
+    } catch (e:any) {
+      const toastMsg = document.getElementById('toast-message');
+      const i = String(e).indexOf('(');
+      if (String(e).includes('bad response')) {
+        toastMsg!.innerHTML = 'The simulation ran into a network error. Please try testing the transaction again.';
+      } else if (String(e).includes(ERROR) && i !== -1) {
+        let errMsg = String(e).substring(ERROR.length, i - 1);
+        errMsg = errMsg[0].toUpperCase().concat(errMsg.substring(1));
+        toastMsg!.innerHTML = errMsg;
+      } else if (String(e).includes(ERROR) && i === -1) {
+        const errMsg = String(e).substring(ERROR.length);
+        toastMsg!.innerHTML = errMsg;
+      } else {
+        toastMsg!.innerHTML = e;
+      }
```

Comment:
You should not be setting innerHTML to a (potentially untrusted) string. Always use `textContent` instead, which allows you to set the text directly. Additionally, use `e.message` as explained above. You can get rid of most of these conditionals that way!